server/applicationServer.go

```diff
@@ -35,9 +37,41 @@ func NewServer() *ApplicationServer {
 // Start will start the server eventually
 func (s *ApplicationServer) Start(addr string) error {
 	s.Config.ServerAddress = addr
-	s.httpServer.Addr = s.Config.ServerAddress
 
-	err := s.httpServer.ListenAndServe()
+	listner, err := net.Listen("tcp", s.Config.ServerAddress)
+	if err != nil {
+		log.Fatal("Listen:", err)
+	} else {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [if block ends with call to log.Fatal function, so drop this else and outdent its block](https://app.codacy.com/gh/hrple/common/pullRequest?prid=5576416)